internal/controllers/topology/cluster/structuredmerge/serversidepathhelper.go

```diff
@@ -70,7 +71,10 @@ func NewServerSidePatchHelper(ctx context.Context, original, modified client.Obj
 
 	// Filter the modifiedUnstructured object to only contain changes intendet to be done.
 	// The originalUnstructured object will be filtered in dryRunSSAPatch using other options.
-	filterObject(modifiedUnstructured, helperOptions)
+	ssa.FilterObject(modifiedUnstructured, &ssa.FilterObjectInput{
```

Comment:
I would suggest to basically replace every occurence of helperOptions with filterOptions but let's please do it in a follow-up PR to keep the diff here smaller.

Reply:
Added a follow-up item in the umbrella issue.